contrib/babelfishpg_tsql/src/session.c

```diff
@@ -156,6 +157,9 @@ set_cur_user_db_and_path(const char *db_name)
 
 	/* set search path */
 	set_search_path_for_user_schema(db_name, user);
+
+	/* set database level collation */
+	set_db_collation_internal(db_id);
```

Comment:
Just wondering, If introducing collation at db-level can affect the output of cross-db queries output. For example, in case of following scenarios:
```
db1 with coll1 -> table t1(int id, varchar a)
db2 with coll2 -> table t2(int id, varchar b)

use db1
select tbl1.a, tbl2.b from t1 as tbl1, db2.t2 as tbl2
```
In above case, will tbl2.b's output will pick up db1's coll? Can it be possible in any cases? If yes, is it expected?

Reply:
Ideally, it should throw an error of collation conflict but its working today. But it should not be blocker. We can create internal issue to track this work. @ahmed-shameem Please file issue for this. 
```
-- inside master whose default collation is SQL_Latin1_General_CP1_CI_As
create table ci_as (a varchar(20))
go

insert into ci_as values ('Dipesh'), ('DiPesh'), ('DIPESH'), ('dipesh')
go

select * from sys.tables where name = 'ci_as'

drop database dipesh_tst

create database dipesh_tst collate SQL_Latin1_General_CP1_CI_AI

use dipesh_tst


create table ci_ai (a varchar(20))
go

insert into ci_ai values ('Dipesh'), ('DiPesh'), ('DIPESH'), ('dipesh')
go

select tbl1.a, tbl2.a from ci_ai tbl1 join master.dbo.ci_as tbl2 on tbl1.a = tbl2.a
```

Reply:
Ack.

Reply:
Other session.c and dbcmds.c changes looks good to me!